cpp/core/operators/shuffle/splitter.cc

```diff
@@ -434,20 +434,25 @@ arrow::Status Splitter::CacheRecordBatch(int32_t partition_id, const arrow::Reco
   int64_t raw_size = batch_nbytes(batch);
   raw_partition_lengths_[partition_id] += raw_size;
   auto payload = std::make_shared<arrow::ipc::IpcPayload>();
+  int64_t batch_compress_time = 0;
 #ifndef SKIPCOMPRESS
   if (batch.num_rows() <= (uint32_t)options_.batch_compress_threshold) {
     TIME_NANO_OR_RAISE(
-        total_compress_time_, arrow::ipc::GetRecordBatchPayload(batch, tiny_bach_write_options_, payload.get()));
+        batch_compress_time, arrow::ipc::GetRecordBatchPayload(batch, tiny_bach_write_options_, payload.get()));
   } else {
     TIME_NANO_OR_RAISE(
-        total_compress_time_, arrow::ipc::GetRecordBatchPayload(batch, options_.ipc_write_options, payload.get()));
+        batch_compress_time, arrow::ipc::GetRecordBatchPayload(batch, options_.ipc_write_options, payload.get()));
   }
 #else
   // for test reason
   TIME_NANO_OR_RAISE(
-      total_compress_time_, arrow::ipc::GetRecordBatchPayload(*batch, tiny_bach_write_options_, payload.get()));
+      batch_compress_time, arrow::ipc::GetRecordBatchPayload(*batch, tiny_bach_write_options_, payload.get()));
 #endif
 
+  if (partition_writer_[partition_id] == nullptr) {
+    partition_writer_[partition_id] = std::make_shared<PartitionWriter>(this, partition_id);
+  }
+  partition_writer_[partition_id]->compress_time += batch_compress_time;
```

Comment:
We needn't record compress time for each partition. A total time is enough. Later if we use async compression by accelerator, we need to add it.